examples/sparse/hetero-rgcn.py

```diff
@@ -0,0 +1,498 @@
+            return self.dropout(h)
+
+        return {ntype: _apply(ntype, h) for ntype, h in hs.items()}
+
+
+
+
+class EntityClassify(nn.Module):
+    def __init__(
+        self,
+        g,
+        h_dim,
+        out_dim,
+        num_bases,
+        num_hidden_layers=1,
+        dropout=0,
+        use_self_loop=False,
+    ):
+        super(EntityClassify, self).__init__()
+        self.g = g
+        self.h_dim = h_dim
+        self.out_dim = out_dim
+        self.rel_names = list(set(g.etypes))
+        self.rel_names.sort()
+        if num_bases < 0 or num_bases > len(self.rel_names):
+            self.num_bases = len(self.rel_names)
+        else:
+            self.num_bases = num_bases
+        self.num_hidden_layers = num_hidden_layers
+        self.dropout = dropout
+        self.use_self_loop = use_self_loop
+
+        self.embed_layer = RelGraphEmbed(g, self.h_dim)
+        self.layers = nn.ModuleList()
+        # i2h
+        self.layers.append(
+            HeteRGConvLayer(
+                self.h_dim,
+                self.h_dim,
+                self.rel_names,
+                self.num_bases,
+                activation=F.relu,
+                self_loop=self.use_self_loop,
+                dropout=self.dropout,
+                weight=False,
+            )
+        )
+        # h2h
+        for i in range(self.num_hidden_layers):
+            self.layers.append(
+                HeteRGConvLayer(
+                    self.h_dim,
+                    self.h_dim,
+                    self.rel_names,
+                    self.num_bases,
+                    activation=F.relu,
+                    self_loop=self.use_self_loop,
+                    dropout=self.dropout,
+                )
+            )
+        # h2o
+        self.layers.append(
+            HeteRGConvLayer(
+                self.h_dim,
+                self.out_dim,
+                self.rel_names,
+                self.num_bases,
+                activation=None,
+                self_loop=self.use_self_loop,
+            )
+        )
+
+    def forward(self, h=None, blocks=None):
+        if h is None:
+            # full graph training
+            h = self.embed_layer()
+        if blocks is None:
+            # full graph training
+            for layer in self.layers:
+                h = layer(self.g, h)
+        else:
+            # minibatch training
+            for layer, block in zip(self.layers, blocks):
+                h = layer(block, h)
+        return h
+
+    def inference(self, g, batch_size, device, num_workers, x=None):
+        """Minibatch inference of final representation over all node types.
+
+        ***NOTE***
+        For node classification, the model is trained to predict on only one node type's
+        label.  Therefore, only that type's final representation is meaningful.
+        """
+
+        if x is None:
+            x = self.embed_layer()
+
+        for l, layer in enumerate(self.layers):
```

Comment:
Regarding this `for` loop body, you can refer to: https://github.com/dmlc/dgl/blob/master/examples/core/graphsage/node_classification.py#L41-L49